### DISCUSSION TITLE
Deploy Hangs at BUILD_ENV

### DISCUSSION THREAD

**Discussion by stewd** [dokku/dokku#5360] [Q&A]
Hi,

I have one server that always hangs when deploying to - Deploy hangs for 20 - 30 seconds just after the BUILD_ENV has been added.

Is there anything happening after the BUILD_ENV that is causing this? Or anywhere specific I can look.

Thanks,

```
Counting objects: 92, done.
Delta compression using up to 8 threads.
Compressing objects: 100% (91/91), done.
Writing objects: 100% (92/92), 14.87 KiB | 1.14 MiB/s, done.
Total 92 (delta 67), reused 0 (delta 0)
-----> Cleaning up...
-----> Building xxxxx from herokuish
-----> Adding BUILD_ENV to build environment...
       BUILD_ENV added successfully
-----> Warning: Multiple default buildpacks reported the ability to handle this app. The first buildpack in the list below will be used.
       Detected buildpacks: ruby nodejs
-----> Ruby app detected

```

**Comment by josegonzalez** [comment]
Can you turn on trace mode and try a deploy? That should show where things are stalling.

**Reply by stewd** [reply]
@josegonzalez  See below. Hangs when running the chown recursive command:

- remote: + local CID=990ba092a6857e681393bb45b674d0082d42b4dee446740697d54f1f4cafcb4f
- remote: + plugn trigger post-container-create app 990ba092a6857e681393bb45b674d0082d42b4dee446740697d54f1f4cafcb4f ayres build
- remote: + source /var/lib/dokku/core-plugins/available/common/functions
- remote: ++ set -eo pipefail
- remote: ++ [[ -n 1 ]]
- remote: ++ set -x
- remote: + [[ ! -n '' ]]
- remote: + docker container start 990ba092a6857e681393bb45b674d0082d42b4dee446740697d54f1f4cafcb4f
- remote: + docker container attach 990ba092a6857e681393bb45b674d0082d42b4dee446740697d54f1f4cafcb4f
- remote: + usermod --home /app herokuishuser
- remote: + chown -R herokuishuser:herokuishuser /app

**Reply by josegonzalez** [reply]
Do you have any volumes mounted into the `/app` directory by any chance?

**Reply by stewd** [reply]
No volumes are mounted for the app.

Memory whilst it hangs looks fine

                            total        used        free      shared  buff/cache   available
              Mem:       24567048     4493828    15550088       27972     4523132    19487440

What I did see though on CPU is "dockerd" spiked to 167% CPU usage for a second

**Reply by josegonzalez** [reply]
You could try switching to `pack` for building your application, which does permissions and such slightly differently (although at the cost of not being able to specify buildpacks in the same way).
